src/Microsoft.DotNet.Interactive/Connection/KernelCommandEnvelope.cs

```diff
@@ -74,7 +74,8 @@ public static void RegisterDefaults()
             [nameof(RequestInput)] = typeof(KernelCommandEnvelope<RequestInput>),
             [nameof(RequestValue)] = typeof(KernelCommandEnvelope<RequestValue>),
             [nameof(RequestValueInfos)] = typeof(KernelCommandEnvelope<RequestValueInfos>),
-            [nameof(RequestKernelInfo)] = typeof(KernelCommandEnvelope<RequestKernelInfo>)
+            [nameof(RequestKernelInfo)] = typeof(KernelCommandEnvelope<RequestKernelInfo>),
+            [nameof(SendValue)] = typeof(KernelCommandEnvelope<SendValue>)
```

Comment:
@jonsequitur The deadlock for value sharing turned out to be same underlying root cause as the `RequestKernelInfo` deadlock that @colombod had fixed in #2788.

There are two related problems that are not fixed in the current PR but would be nice to improve in the future -

1. The error for the missing registration (`KeyNotFoundException`) is being swallowed silently and is not logged anywhere. This was an easy catch once I attached to the `dotnet.exe` process and turned on first-chance exceptions. But considering that we have already stumbled into this twice now, it may be a good idea to log this so that we won't miss this error for new commands introduced in the future. I am not sure if we already have some place in the code where unhandled exceptions can be logged...

2. The reason this issue was not caught by `SerializationTests.All_command_types_are_round_trip_serializable` is because it calls `KernelCommandEnvelope.Create(command)` in the beginning for each command (which registers the command even when it is missing in the above list). We could change the test to check whether each command is registered first (perhaps by creating a new `KernelCommandEnvelope.IsRegistered` API). However, this would also not be 100% fool proof since the dictionaries are static (making it very easy for some unrelated test in the assembly to register the command before `All_command_types_are_round_trip_serializable` is executed).

Another idea may be to avoid the hardcoding in `KernelCommandEnvelope.RegisterDefaults` above and include all well-known commands (and events) by using reflection in the product code. That would potentially address both 1. and 2. above and make this class of bugs impossible to hit...

Reply:
> The error for the missing registration (KeyNotFoundException) is being swallowed silently and is not logged anywhere. 

This was the motivation for adding a diagnostics work item in the March sprint plan. It's worth opening a more specific issue for.

Reply:
I have logged #2884 for the specific issue above. I also jotted down some general logging improvements that could help in #2885.